### ISSUE TITLE
fix: documentation for upgrading Microsoft.Fast.Components.FluentUI 1.6.0 to 2.1.4 Blazor WebAssembly ASP.NET Core Hosted

### ISSUE THREAD

**Issue by Ogglas** [microsoft/fluentui-blazor#363]
# 🐛 Bug Report

Looking at the NuGet for [Microsoft.Fast.Components.FluentUI 2.14][1] :

> When upgrading from an earlier version If you are upgrading from an
> earlier version of the library, please see the [what's new][2] for
> information on (breaking) changes.

The link does not give any more documentation though and only links to the same NuGet.

I have been looking at the latest integration documentation here and there is no mention of the WHATSNEW documentation:

https://learn.microsoft.com/en-us/fluent-ui/web-components/integrations/blazor

Looking at GitHub I did fint the documentation:

https://github.com/microsoft/fluentui-blazor/blob/main/WHATSNEW.md

> An issue has surfaced in **hosted** Blazor projects with version 2.1 and higher. The Client project does not work correctly unless the package is also added to the Server project (without any further changes to the Server's `.csproj` file). We are working together with the AS.NET Core team to solve this. See https://github.com/dotnet/aspnetcore/issues/47135

The error here is _without any further changes to the Server's `.csproj` file_. If you use icons the css will not load for Blazor WebAssembly ASP.NET Core Hosted if you only add a PropertyGroup to the client project. If you add the exact same `PropertyGroup` from [WHATSNEW.md](https://github.com/microsoft/fluentui-blazor/blob/main/WHATSNEW.md) to the server project as well then everything will work.

```
<PropertyGroup>
    <!-- 
        The icon component is part of the library. By default, NO icons (static assets) will be included when publishing the project. 
 
        Setting the property 'PublishFluentIconAssets' to false (default), or leaving the property out completely, will disable publishing of the 
        icon static assets (with exception of the icons that are being used by the library itself). 

        Setting the property 'PublishFluentIconAssets' to 'true' will enable publishing of the icon static assets. You can limit what icon assets get 
        published by specifying a set of icon sizes and a set of variants in the '<FluentIconSizes>' and '<FluentIconVariants>' properties respectively.

        To determine what icons will be published, the specified options for sizes and variants are combined. Specifying sizes '10' and '16' and 
        variants 'Filled' and 'Regular' means all '10/Filled', all '10/Regular', all '16/Filled' and all '16/Regular' icons assets will be published. 
        It is not possible to specify multiple individual combinations like '10/Filled' and '16/Regular' in the same set. 

        When no icon size set is specified in the '<FluentIconSizes>' property, ALL sizes will be included*  
        When no icon variant set is specified in the '<FluentIconVariants>' property, ALL variants will be included*  
        * when publishing of icon assets is enabled 
    -->
    <PublishFluentIconAssets>true</PublishFluentIconAssets>

    <!-- 
        Specify (at least) one or more sizes from the following options (separated by ','):
        10,12,16,20,24,28,32,48 
        Leave out the property to have all sizes included.
    -->
    <FluentIconSizes>10,12,16,20,24,28,32,48</FluentIconSizes>

    <!-- 
        Specify (at least) one or more variants from the following options (separated by ','):
        Filled,Regular 
        Leave out the property to have all variants included.
    -->
    <FluentIconVariants>Filled,Regular</FluentIconVariants>

    <!-- 
        The emoji component is part of the library. By default, NO emojis (static assets) will be included when publishing the project. 
 
        Setting the property 'PublishFluentEmoji' to false (default), or leaving the property out completely, will disable publishing of the emoji static assets. 

        Setting the property 'PublishFluentEmojiAssets' to 'true' will enable publishing of the emoji static assets. You can limit what emoji assets get 
        published by specifying a set of emoji groups and a set of emoji styles in the '<FluentEmojiGroups>' and '<FluentEmojiStyles>' properties respectively.

        To determine what emojis will be published, the specified options for sizes and variants are combined. Specifying emoji groups 'Activities' and 'Flags' 
        and emoji styles 'Color' and 'Flat' means all 'Activities/Color', all 'Activities/Flat', all 'Flags/Color' and all 'Flags/Flat' emoji assets will be published.

        It is not possible to specify multiple individual combinations like 'Activities/Color' and 'Flags/Flat' in the same published set

        When no emoji group set is specified in the '<FluentEmojiGroups>' property, ALL groups will be included*  
        When no emoji variant set is specified in the '<FluentEmojiStyles>' property, ALL styles will be included*  
        * when publishing of emoji assets is enabled 
    -->
    <PublishFluentEmojiAssets>true</PublishFluentEmojiAssets>

    <!-- 
        Specify (at least) one or more groups from the following options (separated by ','):
        Activities,Animals_Nature,Flags,Food_Drink,Objects,People_Body,Smileys_Emotion,Symbols,Travel_Places 
        Leave out the property to have all groups included.
    -->
    <FluentEmojiGroups>Activities,Animals_Nature,Flags,Food_Drink,Objects,People_Body,Smileys_Emotion,Symbols,Travel_Places</FluentEmojiGroups>

    <!-- 
        Specify (at least) one or more styles from the following options (separated by ','): 
        Color,Flat,HighContrast
        Leave out the property to have all styles included.
    -->
    <FluentEmojiStyles>Color,Flat,HighContrast</FluentEmojiStyles>
</PropertyGroup>
```

Example error from console if only `PropertyGroup` is added to the client project and then trying to load an icon:

https://localhost:44397/_content/Microsoft.Fast.Components.FluentUI/icons/Delete/16_f.svg 404

If only Github documentation is updated I think you should skip the other documentation and just point everyone here.

Another thing is that it is now hard to find examples for Microsoft.Fast.Components.FluentUI 1.6.0 for example. It would be nice if the component library site was versioned so that also old versions could be viewed. 

Not related but it would also be nice with a simpler base URL than https://brave-cliff-0c0c93310.azurestaticapps.net/ for the official samples. I know https://aka.ms/fluentui-blazor works but it would be nice with a more friendly URL. :)

Original issue from SO:

https://stackoverflow.com/questions/75938834/upgrading-microsoft-fast-components-fluentui-1-6-0-to-2-1-4-blazor-webassembly-d

  [1]: https://www.nuget.org/packages/Microsoft.Fast.Components.FluentUI/2.1.4#readme-body-tab
  [2]: https://www.nuget.org/packages/Microsoft.Fast.Components.FluentUI/2.1.4

**Comment by vnbaaij** [comment]
Hi Oscar,

NuGet uses the same readme as the repo. Obviously, the link only works in the repo then. I changed it now so for the next version the WhatsNew link on NuGet (and in the readme) will point to https://aka.ms/fluentui-blazor/whatsnew.

I'll make a pr for the integration docs to point to the `whatsnew` as well

I haven't experienced needing to add the `<PropertyGroup>` to the server project in a hosted setup. I'll investigate further,

> If only Github documentation is updated I think you should skip the other documentation and just point everyone here.
We will probably remove the integration docs in the future because of some changes we have in mind.

>Another thing is that it is now hard to find examples for Microsoft.Fast.Components.FluentUI 1.6.0 for example. It would be >nice if the component library site was versioned so that also old versions could be viewed.
Not seeing this happening any time soon, I'm afraid

>Not related but it would also be nice with a simpler base URL than https://brave-cliff-0c0c93310.azurestaticapps.net/ for the >official samples. I know https://aka.ms/fluentui-blazor works but it would be nice with a more friendly URL. :)

We are working on this...stay tuned

**Comment by Ogglas** [comment]
Hello again Vincent,

Sounds good with the readme and integration docs! One idea is simply having a link to Github in integration docs. I created a PR for WHATSNEW.md with some formatting https://github.com/microsoft/fluentui-blazor/pull/364

Do that! I tested it again. Using Chrome I used DevTools -> Application -> Storage and then Clear site data. After that I needed `<PropertyGroup>` in both client and server to get Icons working. 

Tags can always be used even if it is not as efficient. :)

https://github.com/microsoft/fluentui-blazor/blob/V1.6.0/examples/FluentUI.Demo.Shared/Pages/Select/Examples/SelectFromList.razor
